pytensor/tensor/shape.py

```diff
@@ -994,6 +995,9 @@ def specify_broadcastable(x, *axes):
     if not axes:
         return x
 
+    if len(axes) == 1 and isinstance(axes[0], Collection):
+        axes = axes[0]
+
```

Comment:
Why is this needed?
```suggestion
```

Reply:
the definition of specify_broadcastable is:
```
def specify_broadcastable(x, *axes):
```
The ```*``` converts axes into an iterable like (axes, ). Hence when we expect our non_broadcastable_axis list to be let's say [0], specify_broadcastable actually interprets it as ([0], ).

The comparison operation below hence throws an error:
https://github.com/pymc-devs/pytensor/blob/bcd81c7830d1ce0ccb2c16ac7010d23d1fcaafc3/pytensor/tensor/shape.py#L997

Reply:
You can unpack the axis when you call specify_broadcastable woth *

```python
axes = [0, 1, 2]
pt.specify_broadcastable(x, *axes)
```

Works the same way regardless of whether there are multiple or a single axis